### ISSUE TITLE
"A request has failed" in VS Code extension

### ISSUE THREAD

**Issue by letmaik** [microsoft/pyright#633]
**Describe the bug**
I started editing Python files and suddenly "A request has failed" errors popped up from pyright. 
![grafik](https://user-images.githubusercontent.com/530988/79851989-faeb3e80-83bd-11ea-926a-283e0b074814.png)

**To Reproduce**
Not sure what triggered it. It's the first time I see it, normally the extension is very reliable. I can't reproduce it anymore, but the error happened 4-5 times in a row, before stopping.

**Expected behavior**
Should not crash but even if, then it should be possible to hide the error messages instead of popping them up again and again.

**Screenshots or Code**

```
Searching for source files
Found 46 source files
events.js:191
    throw err; // Unhandled 'error' event
    ^

Error [ERR_UNHANDLED_ERROR]: Unhandled error. ('DOMException [DataCloneError]: Error: request cancelled could not be ' +
  'cloned.\n    at h ' +
  '(/home/user/.vscode-server-insiders/extensions/ms-pyright.pyright-1.1.33/server/server.bundle.js:21:26592)\n' +
  '    at MessagePort.<anonymous> ' +
  '(/home/user/.vscode-server-insiders/extensions/ms-pyright.pyright-1.1.33/server/server.bundle.js:21:28291)\n' +
  '    at MessagePort.emit (events.js:200:13)\n    at MessagePort.onmessage ' +
  '(internal/worker/io.js:70:8)')
    at Worker.emit (events.js:189:17)
    at Worker.[kOnErrorMessage] (internal/worker.js:176:10)
    at Worker.[kOnMessage] (internal/worker.js:186:37)
    at MessagePort.<anonymous> (internal/worker.js:118:57)
    at MessagePort.emit (events.js:200:13)
    at MessagePort.onmessage (internal/worker/io.js:70:8) {
  context: 'DOMException [DataCloneError]: Error: request cancelled could not be ' +
    'cloned.\n    at h ' +
    '(/home/user/.vscode-server-insiders/extensions/ms-pyright.pyright-1.1.33/server/server.bundle.js:21:26592)\n' +
    '    at MessagePort.<anonymous> ' +
    '(/home/user/.vscode-server-insiders/extensions/ms-pyright.pyright-1.1.33/server/server.bundle.js:21:28291)\n' +
    '    at MessagePort.emit (events.js:200:13)\n    at MessagePort.onmessage ' +
    '(internal/worker/io.js:70:8)'
}
[Info  - 10:15:08 AM] Connection to server got closed. Server will restart.
[Error - 10:15:08 AM] Request textDocument/codeAction failed.
Error: Connection got disposed.
	at Object.dispose (/home/user/.vscode-server-insiders/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-jsonrpc/lib/main.js:967:25)
	at Object.dispose (/home/user/.vscode-server-insiders/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-languageclient/lib/client.js:74:35)
	at LanguageClient.handleConnectionClosed (/home/user/.vscode-server-insiders/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-languageclient/lib/client.js:2320:42)
	at LanguageClient.handleConnectionClosed (/home/user/.vscode-server-insiders/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-languageclient/lib/main.js:155:15)
	at closeHandler (/home/user/.vscode-server-insiders/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-languageclient/lib/client.js:2307:18)
	at CallbackList.invoke (/home/user/.vscode-server-insiders/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-jsonrpc/lib/events.js:62:39)
	at Emitter.fire (/home/user/.vscode-server-insiders/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-jsonrpc/lib/events.js:125:36)
	at closeHandler (/home/user/.vscode-server-insiders/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-jsonrpc/lib/main.js:288:26)
	at CallbackList.invoke (/home/user/.vscode-server-insiders/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-jsonrpc/lib/events.js:62:39)
	at Emitter.fire (/home/user/.vscode-server-insiders/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-jsonrpc/lib/events.js:125:36)
	at IPCMessageReader.fireClose (/home/user/.vscode-server-insiders/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-jsonrpc/lib/messageReader.js:111:27)
	at ChildProcess.<anonymous> (/home/user/.vscode-server-insiders/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-jsonrpc/lib/messageReader.js:213:45)
	at ChildProcess.emit (events.js:205:15)
	at maybeClose (internal/child_process.js:1021:16)
	at Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)
[Error - 10:15:08 AM] Request textDocument/documentSymbol failed.
Error: Connection got disposed.
	at Object.dispose (/home/user/.vscode-server-insiders/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-jsonrpc/lib/main.js:967:25)
	at Object.dispose (/home/user/.vscode-server-insiders/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-languageclient/lib/client.js:74:35)
	at LanguageClient.handleConnectionClosed (/home/user/.vscode-server-insiders/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-languageclient/lib/client.js:2320:42)
	at LanguageClient.handleConnectionClosed (/home/user/.vscode-server-insiders/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-languageclient/lib/main.js:155:15)
	at closeHandler (/home/user/.vscode-server-insiders/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-languageclient/lib/client.js:2307:18)
	at CallbackList.invoke (/home/user/.vscode-server-insiders/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-jsonrpc/lib/events.js:62:39)
	at Emitter.fire (/home/user/.vscode-server-insiders/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-jsonrpc/lib/events.js:125:36)
	at closeHandler (/home/user/.vscode-server-insiders/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-jsonrpc/lib/main.js:288:26)
	at CallbackList.invoke (/home/user/.vscode-server-insiders/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-jsonrpc/lib/events.js:62:39)
	at Emitter.fire (/home/user/.vscode-server-insiders/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-jsonrpc/lib/events.js:125:36)
	at IPCMessageReader.fireClose (/home/user/.vscode-server-insiders/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-jsonrpc/lib/messageReader.js:111:27)
	at ChildProcess.<anonymous> (/home/user/.vscode-server-insiders/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-jsonrpc/lib/messageReader.js:213:45)
	at ChildProcess.emit (events.js:205:15)
	at maybeClose (internal/child_process.js:1021:16)
	at Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)
Pyright language server starting
Server root directory: /home/user/.vscode-server-insiders/extensions/ms-pyright.pyright-1.1.33
```

**VS Code extension or command-line**
VS Code

**Additional context**

**Comment by erictraut** [comment]
@heejaechang This looks related to the background analysis worker process code that you recently added. I don't yet understand the root cause. Any thoughts?

**Comment by rsokl** [comment]
I'm hitting the same issue. It looks like there is an attempt to set `typingsPath` to an invalid directory

```
Setting pythonPath for service "MyGrad": "/home/user/anaconda3/envs/hardnet/bin/python"
typingsPath /home/user/MyGrad/typings is not a valid directory.
Searching for source files
Found 36 source files
events.js:191
    throw err; // Unhandled 'error' event
    ^

Error [ERR_UNHANDLED_ERROR]: Unhandled error. ('DOMException [DataCloneError]: Error: request cancelled could not be ' +
  'cloned.\n    at h ' +
  '(/home/user/.vscode-server/extensions/ms-pyright.pyright-1.1.33/server/server.bundle.js:21:26592)\n' +
  '    at MessagePort.<anonymous> ' +
  '(/home/user/.vscode-server/extensions/ms-pyright.pyright-1.1.33/server/server.bundle.js:21:28291)\n' +
  '    at MessagePort.emit (events.js:200:13)\n    at ' +
  'MessagePort.onmessage (internal/worker/io.js:70:8)')
    at Worker.emit (events.js:189:17)
    at Worker.[kOnErrorMessage] (internal/worker.js:176:10)
    at Worker.[kOnMessage] (internal/worker.js:186:37)
    at MessagePort.<anonymous> (internal/worker.js:118:57)
    at MessagePort.emit (events.js:200:13)
    at MessagePort.onmessage (internal/worker/io.js:70:8) {
  context: 'DOMException [DataCloneError]: Error: request cancelled could not be ' +
    'cloned.\n    at h ' +
    '(/home/user/.vscode-server/extensions/ms-pyright.pyright-1.1.33/server/server.bundle.js:21:26592)\n' +
    '    at MessagePort.<anonymous> ' +
    '(/home/user/.vscode-server/extensions/ms-pyright.pyright-1.1.33/server/server.bundle.js:21:28291)\n' +
    '    at MessagePort.emit (events.js:200:13)\n    at ' +
    'MessagePort.onmessage (internal/worker/io.js:70:8)'
}
[Info  - 11:02:42 AM] Connection to server got closed. Server will restart.
[Error - 11:02:42 AM] Request textDocument/codeAction failed.
Error: Connection got disposed.
	at Object.dispose (/home/user/.vscode-server/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-jsonrpc/lib/main.js:967:25)
	at Object.dispose (/home/user/.vscode-server/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-languageclient/lib/client.js:74:35)
	at LanguageClient.handleConnectionClosed (/home/user/.vscode-server/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-languageclient/lib/client.js:2320:42)
	at LanguageClient.handleConnectionClosed (/home/user/.vscode-server/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-languageclient/lib/main.js:155:15)
	at closeHandler (/home/user/.vscode-server/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-languageclient/lib/client.js:2307:18)
	at CallbackList.invoke (/home/user/.vscode-server/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-jsonrpc/lib/events.js:62:39)
	at Emitter.fire (/home/user/.vscode-server/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-jsonrpc/lib/events.js:125:36)
	at closeHandler (/home/user/.vscode-server/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-jsonrpc/lib/main.js:288:26)
	at CallbackList.invoke (/home/user/.vscode-server/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-jsonrpc/lib/events.js:62:39)
	at Emitter.fire (/home/user/.vscode-server/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-jsonrpc/lib/events.js:125:36)
	at IPCMessageReader.fireClose (/home/user/.vscode-server/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-jsonrpc/lib/messageReader.js:111:27)
	at ChildProcess.<anonymous> (/home/user/.vscode-server/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-jsonrpc/lib/messageReader.js:213:45)
	at ChildProcess.emit (events.js:205:15)
	at maybeClose (internal/child_process.js:1021:16)
	at Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)
[Error - 11:02:42 AM] Request textDocument/codeAction failed.
Error: Connection got disposed.
	at Object.dispose (/home/user/.vscode-server/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-jsonrpc/lib/main.js:967:25)
	at Object.dispose (/home/user/.vscode-server/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-languageclient/lib/client.js:74:35)
	at LanguageClient.handleConnectionClosed (/home/user/.vscode-server/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-languageclient/lib/client.js:2320:42)
	at LanguageClient.handleConnectionClosed (/home/user/.vscode-server/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-languageclient/lib/main.js:155:15)
	at closeHandler (/home/user/.vscode-server/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-languageclient/lib/client.js:2307:18)
	at CallbackList.invoke (/home/user/.vscode-server/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-jsonrpc/lib/events.js:62:39)
	at Emitter.fire (/home/user/.vscode-server/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-jsonrpc/lib/events.js:125:36)
	at closeHandler (/home/user/.vscode-server/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-jsonrpc/lib/main.js:288:26)
	at CallbackList.invoke (/home/user/.vscode-server/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-jsonrpc/lib/events.js:62:39)
	at Emitter.fire (/home/user/.vscode-server/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-jsonrpc/lib/events.js:125:36)
	at IPCMessageReader.fireClose (/home/user/.vscode-server/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-jsonrpc/lib/messageReader.js:111:27)
	at ChildProcess.<anonymous> (/home/user/.vscode-server/extensions/ms-pyright.pyright-1.1.33/node_modules/vscode-jsonrpc/lib/messageReader.js:213:45)
	at ChildProcess.emit (events.js:205:15)
	at maybeClose (internal/child_process.js:1021:16)
	at Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)
Pyright language server starting
Server root directory: /home/user/.vscode-server/extensions/ms-pyright.pyright-1.1.33
Starting service instance "MyGrad"
Received updated settings
No configuration file found.
Setting pythonPath for service "MyGrad": "/home/user/anaconda3/envs/hardnet/bin/python"
typingsPath /home/user/MyGrad/typings is not a valid directory.
Searching for source files
Found 36 source files
Background analysis root directory: /home/user/.vscode-server/extensions/ms-pyright.pyright-1.1.33
Background analysis started
```

**Comment by erictraut** [comment]
Thanks for the additional information. The typingsPath log entry isn't related to the crash. 

Could you tell me which version of VS Code you are running? Is it the latest available (1.44.2)?

**Comment by rsokl** [comment]
Yep, VS Code version 1.44.2

**Comment by heejaechang** [comment]
Let me make it not to crash first.

**Comment by erictraut** [comment]
Thanks again for the bug reports. This is now fixed in 1.1.34, which I just published.
